### PR TITLE
Completed the endpoint for articles/:article_id/comments - Ticket #15

### PR DESCRIPTION
**Pull request by ActuallySilver**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -138,6 +137,28 @@ describe("/api", () => {
             });
         });
       });
+      describe("/comments", () => {
+        describe("GET", () => {
+          test("200 - recieves back a list of comments on the article", () => {
+            return request(app)
+              .get("/api/articles/1/comments")
+              .expect(200)
+              .then(({ body: { comments } }) => {
```
Comment (Poonam-raj) [inline __tests__/app.test.js:146]: would like to see a comments.length check to make sure you're getting the right number of comments back

File: __tests__/app.test.js [hunk 2]
```diff
@@ -138,6 +137,28 @@ describe("/api", () => {
...
+                comments.forEach((comment) => {
+                  expect(comment).toEqual(
+                    expect.objectContaining({
+                      body: expect.any(String),
+                      votes: expect.any(Number),
+                      author: expect.any(String),
+                      comment_id: expect.any(Number),
+                      created_at: expect.stringMatching(/(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}.\d{3}Z)/),
+                    })
+                  );
```
Comment (Poonam-raj) [inline __tests__/app.test.js:157]: Missing tests:

- If valid and existing article doesn't have any comments
- If valid article ID but not existing article is requested

File: app/controllers/comments.controller.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const models = require("../models");
+exports.getCommentsByArticleID = async (req, res, next) => {
+  try {
+    const { article_id: articleId } = req.params;
+    const [comments] = await Promise.all([
+      models.comments.selectCommentsByArticleID(articleId),
```
Comment (Poonam-raj) [inline app/controllers/comments.controller.js:6]: Why are you doing the selectArticleById here?

**Review by Poonam-raj — CHANGES_REQUESTED** [review]
Few missing tests here, also question on the controller set up

File: app/controllers/comments.controller.js (hunk 3 above)
Comment (ActuallySilver) [reply app/controllers/comments.controller.js:6]: Returns an error if the article doesn't exist, basically just a check if the article exists

**Review by Angaca — APPROVED** [review]
Ready to go

File: app/controllers/comments.controller.js (hunk 3 above)
Comment (Poonam-raj) [reply app/controllers/comments.controller.js:6]: Makes sense ty!

**Pull request merged**